In [36]:
import numpy as np

# Домашнее задание к уроку 5


## Задание 1

Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

In [37]:
M = 80
n = 256
std = 16
p = 0.95
alpha = 1 - p

In [38]:
# Используем z-критерий, поскольку среднеквадратичное отклонение известно
# Из таблицы получаем значение z для alpha/2:
print(alpha / 2)
z = 1.96

0.025000000000000022


In [39]:
# Стандартная ошибка среднего
SE = std / n**0.5

In [40]:
# Левая граница интервала
int_left = M - z * SE
int_left

78.04

In [41]:
# Правая граница интервала
int_right = M + z * SE
int_right

81.96

**Ответ**: Доверительный интервал для оценки математического ожидания с надёжностью 0.95: $[78,04;81,96]$

## Задание 2

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [42]:
n = 10
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
p = 0.95
alpha = 1 - p

In [43]:
# Вычислим математическое ожидание для выборки
M = np.mean(x)
M

6.590000000000001

In [44]:
# Вычислим несмещённое среднеквадратичное отклонение
std = np.std(x, ddof = 1)
std

0.4508017549014448

In [45]:
# Используем критерий Стьюдента, поскольку среднеквадратичное отклонение неизвестно
# Из таблицы получаем значение t для alpha/2 и 9 степеней свободы:
print(alpha / 2)
t = 2.262

0.025000000000000022


In [46]:
# Стандартная ошибка среднего
SE = std / n**0.5

In [47]:
# Левая граница интервала
int_left = M - t * SE
int_left

6.267538255912426

In [48]:
# Правая граница интервала
int_right = M + t * SE
int_right

6.912461744087575

**Ответ:** Доверительный интервал для величины $X$ с вероятностью 0.95: $[6,267;6,912]$

## Задание 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

In [49]:
mu_0 = 17
mu = 17.5
alpha = 0.05
n = 100
D = 4

Гипотеза $H_0: \mu = \mu_0$<br>
Гипотеза $H_1: \mu > \mu_0$

In [50]:
# Используем z-критерий, поскольку среднеквадратичное отклонение известно
# Из таблицы для 0.95 получаем значение z
z = 1.65

In [51]:
# Стандартная ошибка
SE = D**0.5 / n**0.5

In [52]:
(mu - mu_0) / SE

2.5

Т.к. 2,5 > 1,65, то верна гипотеза $H_1$

**Ответ:** Гипотеза $H_1$ верна на уровне значимости 0,95

---



## Задание 4

Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [53]:
mu_0 = 200
alpha = 0.01
n = 10
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

Гипотеза $H_0: \mu = \mu_0$<br>
Гипотеза $H_1: \mu \neq \mu_0$

In [54]:
# Используем критерий Стьюдента, поскольку среднеквадратичное отклонение неизвестно
# Из таблицы получаем значение t для alpha/2 и 9 степеней свободы:
print(alpha / 2)
t = 3.25

0.005


In [55]:
# Вычислим математическое ожидание для выборки
mu = np.mean(x)
mu

198.5

In [56]:
# Вычислим несмещённое среднеквадратичное отклонение
std = np.std(x, ddof = 1)
std

4.453463071962462

In [57]:
# Стандартная ошибка среднего
SE = std / n**0.5

In [58]:
(mu - mu_0) / SE

-1.0651074037450896

Т.к. -1.065 попадает в интервал от -3.25 до 3.25, то гипотеза $H_0$ верна

**Ответ:** Гипотеза $H_0$ верна c доверительной вероятностью 0,99


